src/examples/crypto/ecdsa/run.ts

```diff
@@ -34,3 +41,42 @@ console.timeEnd('keccak + ecdsa verify (prove)');
 
 proof.publicOutput.assertTrue('signature verifies');
 assert(await keccakAndEcdsa.verify(proof), 'proof verifies');
+
+// Hardcoded ethers.js signature and inputs for verification in o1js
+
+// message signed using ethers.js
+const msg = 'Secrets hidden, truth in ZKPs ;)';
+
+// uncompressed public key generated by ethers.js
+const uncompressedPublicKey =
+  '0x040957928494c38660d254dc03ba78f091a4aea0270afb447f193c4daf6648f02b720071af9b5bda4936998ec186e632f4be82886914851d7c753747b0a949d1a4';
+
+// compressed public key generated by ethers.js
+const compressedPublicKey =
+  '0x020957928494c38660d254dc03ba78f091a4aea0270afb447f193c4daf6648f02b';
+
+// ECDSA signature generated by ethers.js
+const rawSignature =
+  '0x6fada464c3bc2ae127f8c907c0c4bccbd05ba83a584156edb808b7400346b4c9558598d9c7869f5fd75d81128711f6621e4cb5ba2f52a2a51c46c859f49a833a1b';
+
+const publicKeyE = Secp256k1.fromEthers(compressedPublicKey.slice(2));
```

Comment:
you do `.slice(2)` on the input from ethers.js here

couldn't it be a source of confusion that the input can't be passed in directly? maybe `fromEthers` should look at the first two chars and handle the case when they are "0x"

Reply:
yes, I'll add a check to handle both cases: with and without "0x"

Reply:
I remember now that I didn’t fix it as you suggested initially because I wanted to keep it consistent with the `Bytes.fromHex` API. I've now addressed it with the following:


```ts
static fromHex(hex: string) {
    // trim the '0x' prefix if present
    if (hex.startsWith('0x')) {
      hex = hex.slice(2);
     }
     // ...
}
```

@mitschabaude Do you think it’s worth tweaking `Bytes.fromHex` in a similar way as well?

Reply:
hm your call. I think it's fine for `fromEthers()` which takes a string formatted in a particular library's style, but I don't think we need it in our general `Bytes.fromHex()` method 🤔